# hunk 1: multi-inheritance is broken https://sourceforge.net/p/swig/bugs/1379/
# hunk 2: compilation issue on GCC 4.6 http://sourceforge.net/p/swig/patches/296/
# hunk 3: thread bug on iterators https://sourceforge.net/p/swig/bugs/1370/
# hunk 4: thread bug on virtual methods https://github.com/swig/swig/issues/306
--- DAsdk_wrap.cpp-t
+++ DAsdk_wrap.cpp-t
@@ -1618,6 +1618,10 @@
   if (!SwigPyObject_Check(next)) {
     return NULL;
   }
+  while (sobj->next && SwigPyObject_Check(sobj->next)) {
+     sobj = (SwigPyObject *)sobj->next;
+  }
+  ((SwigPyObject *)next)->next = sobj->next;
   sobj->next = next;
   Py_INCREF(next);
   return SWIG_Py_Void();
@@ -3492,6 +3496,7 @@
 
 #include <string>
 #include <stdexcept>
+#include <cstddef>
 
   
 namespace swig {
@@ -3508,7 +3513,11 @@
     }
       
   public:
-    virtual ~SwigPyIterator() {}
+    virtual ~SwigPyIterator() {
+       SWIG_PYTHON_THREAD_BEGIN_BLOCK;
+       _seq = 0;
+       SWIG_PYTHON_THREAD_END_BLOCK;
+    }
 
     // Access iterator method, required by Python
     virtual PyObject *value() const = 0;
@@ -33457,11 +33466,13 @@
   try {
     {
       try {
+        SWIG_PYTHON_THREAD_BEGIN_ALLOW;
         if (upcall) {
           (arg1)->DA::sdk::main_loop(arg2,arg3);
         } else {
           (arg1)->main_loop(arg2,arg3);
         }
+        SWIG_PYTHON_THREAD_END_ALLOW;
       } catch (error const & e) {
         throw_py_error(e);
         SWIG_fail;
